src/formatDate.js

```diff
@@ -50,7 +50,70 @@
  */
 
 function formatDate(date, fromFormat, toFormat) {
-  // write code here
+  let [newDay, newMonth, newYear] = '';
+  let n = 0;
+  let result = '';
+  let shortNewYear = '';
+  let shortOldFormat = '';
+  let longNewYear = '';
+
+  for (const oldIndexDate of fromFormat) {
+    if (oldIndexDate === 'MM') {
+      newMonth = date.slice(n, (n + 2));
+      n = n + 3;
+    }
+
+    if (oldIndexDate === 'YY') {
+      newYear = date.slice(n, (n + 2));
+      shortOldFormat = 'yes';
+      n = n + 3;
+
+      if (+newYear < 30) {
+        longNewYear = '20' + newYear;
+      } else {
+        longNewYear = '19' + newYear;
+      }
```

Comment:
- 20, 19, 30 are the magic numbers, so make sense to move them to constants 
- make sense to use ternary operator as it will make a code shorter 
Check other places